From 452101a4d549e580bcf17e4b6ef053e968a52c2b Mon Sep 17 00:00:00 2001
From: John Howard <howardjohn@google.com>
Date: Wed, 3 Nov 2021 19:34:57 -0700
Subject: Cleanup duplicate pilot SAN log (#35874)

This function was called in 2 places so we got 2x logs which looked like
a bug. So just clean it up
---
 pkg/istio-agent/agent.go | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/pkg/istio-agent/agent.go b/pkg/istio-agent/agent.go
index cf4ec0cd82..4ded6de207 100644
--- a/pkg/istio-agent/agent.go
+++ b/pkg/istio-agent/agent.go
@@ -237,7 +237,6 @@ func (a *Agent) generateNodeMetadata() (*model.Node, error) {
 		// Obtain Pilot SAN, using DNS.
 		pilotSAN = []string{config.GetPilotSan(a.proxyConfig.DiscoveryAddress)}
 	}
-	log.Infof("Pilot SAN: %v", pilotSAN)
 
 	return bootstrap.GetNodeMetaData(bootstrap.MetadataOptions{
 		ID:                  a.cfg.ServiceNode,
@@ -260,6 +259,8 @@ func (a *Agent) initializeEnvoyAgent(ctx context.Context) error {
 		return fmt.Errorf("failed to generate bootstrap metadata: %v", err)
 	}
 
+	log.Infof("Pilot SAN: %v", node.Metadata.PilotSubjectAltName)
+
 	// Note: the cert checking still works, the generated file is updated if certs are changed.
 	// We just don't save the generated file, but use a custom one instead. Pilot will keep
 	// monitoring the certs and restart if the content of the certs changes.
-- 
2.35.3

